lib/src/entrypoint.dart

```diff
@@ -916,6 +931,113 @@ See https://dart.dev/go/sdk-constraint
     dataError('The "$packageConfigPath" file is not recognized by '
         '"pub" version, please run "$topLevelProgram pub get".');
   }
+
+  /// Looks through the root-[incompability] of a solve-failure and tries to see
+  /// if any single change can make a resolution go through. Returns a formatted
+  /// list of suggestions, or the empty String if no suggestions were found.
+  Future<String> _suggestResolutionAlternatives(SolveType type,
+      Incompatibility incompatibility, Iterable<String> unlock) async {
+    final b = StringBuffer();
+    final visited = <String>{};
+    final stopwatch = Stopwatch()..start();
+    var first = true;
+    void suggest(String suggestion) {
+      if (first) {
+        b.writeln(
+            '\nYou can try one of the following suggestion(s) to resolve the failure:');
```

Comment:
nit: Can't we make `(s)` be there or not...

Reply:
Done